src/aics-image-viewer/components/App/index.tsx

```diff
@@ -420,12 +432,32 @@ const App: React.FC<AppProps> = (props) => {
 
     const channelNames = aimg.imageInfo.channelNames;
     const newChannelSettings = setChannelStateForNewImage(channelNames);
+
+    // order is important:
+    // we need to remove the old volume before triggering channels unloaded,
+    // which may cause calls on View3d to the old volume.
+    view3d.removeAllVolumes();
     setAllChannelsUnloaded(channelNames.length);
+    placeImageInViewer(aimg, newChannelSettings);
     channelRangesRef.current = new Array(channelNames.length).fill(undefined);
 
     const requiredLoadSpec = new LoadSpec();
     requiredLoadSpec.time = viewerState.current.time;
 
+    // make the currently enabled channels "required":
+    // find all enabled indices in newChannelSettings:
+    const requiredChannelsToLoad = newChannelSettings
+      ? newChannelSettings.map((channel, index) => (channel.volumeEnabled ? index : -1)).filter((index) => index >= 0)
+      : [];
+
+    // add mask channel to required channels?
+    const maskChannelName = getCurrentViewerChannelSettings()?.maskChannelName || "";
+    const maskChannelIndex = channelNames.indexOf(maskChannelName);
+    if (maskChannelIndex >= 0 && !requiredChannelsToLoad.includes(maskChannelIndex)) {
+      requiredChannelsToLoad.push(maskChannelIndex);
+    }
+    requiredLoadSpec.channels = requiredChannelsToLoad;
+
```

Comment:
the above set of lines 447-460 actually explicitly selects channels to load.  I'm not certain of whether this is necessary, will test one more time without it. I think it makes a difference when switching images, especially if we fail to identify the mask channel.

Reply:
I did test this and I think it makes a minor difference in channel load order but we still get correct results either way.  I am inclined to leave it in.  The only scenario I would worry about is the possibility of loading only a mask channel and no others but it would be hard to unintentionally get a data scenario where that happens.